### PR TITLE
Add Spack to the tool suggestions for scientific computing

### PR DESCRIPTION
**Pull request by tgamblin**
- Added link to Spack.
- Fixed references to hashdist and conda.

### CODE REVIEW

File: source/current.rst [hunk 1]
```diff
@@ -23,9 +23,14 @@ Installation Tool Recommendations
   dependencies from a shared Python installation. [4]_
 
 * If you're looking for management of fully integrated cross-platform software
-  stacks, consider :ref:`buildout` (primarily focused on the web development
-  community) or :ref:`hashdist`, or :ref:`conda` (both primarily focused on
-  the scientific community).
+  stacks, consider:
+  
+  * `buildout <http://www.buildout.org/>`_ (primarily focused on the web development community)
```
Comment (ncoghlan) [inline source/current.rst]: These "ref" links are to the internal list of projects at https://packaging.python.org/projects/ but it looks like that isn't building on ReadTheDocs for some reason (source link is at https://github.com/pypa/python-packaging-user-guide/blob/master/source/projects.rst ).

**Comment by ncoghlan** [comment]
I opened https://github.com/pypa/python-packaging-user-guide/issues/253 to cover the project summary page being broken.

**Comment by tgamblin** [comment]
@ncoghlan: so should I close this or keep it open?  The idea was to get a reference to [Spack](http://github.com/LLNL/spack) into the tool recommendations for scientific computing.

**Comment by ncoghlan** [comment]
If you rebase on master, you'll get the changes I just made in #254 that should fix the cross-reference resolution problems.

That will let you add Spack to the list of Non-PyPA projects, and then adjust the scientific Python page accordingly.

**Comment by ncoghlan** [comment]
As far as Spack not being in PyPI itself goes, that would be worth mentioning in the project summary.

**Comment by tgamblin** [comment]
Done -- how does that look?

**Comment by ncoghlan** [comment]
Merged, thanks! You may also be interested to know I added it to my list at https://fedoraproject.org/wiki/Env_and_Stacks/Projects/UserLevelPackageManagement to take a closer look at in the event I ever get back to working on that project :)

**Comment by tgamblin** [comment]
@ncoghlan: Thanks!  You might be interested to know we've been working with some folks at RedHat using Spack and [libabigail](http://www.sourceware.org/libabigail/) to test ABI compatibility between compilers.  Ben Woodard is working on that.  Definitely let me know if you get back to your project -- HPC people have all the problems described on that page.

**Comment by IanLee1521** [comment]
@ncoghlan -- Is there an issue with the ReadTheDocs not rebuilding? Looks like https://packaging.python.org/current/ is missing this change. Cheers!

**Comment by ncoghlan** [comment]
@IanLee1521 I'm not sure - even though the RTD dashboard reported a successful build, the change wasn't displaying for me either. However, I just triggered a new build to fix a Sphinx cross-reference warning, and it's rendering properly now.

@tgamblin You may also want to add a section on Spack to https://packaging.python.org/science/

**Comment by tgamblin** [comment]
@ncoghlan: Thanks. Submitted!

**Pull request merged**